src/Integration.Vsix/Delegates/SonarAnalyzerManager.cs

```diff
@@ -0,0 +1,135 @@
+            }
+            if (qualityProfileProvider == null)
+            {
+                throw new ArgumentNullException(nameof(qualityProfileProvider));
+            }
+            if (vsSolution == null)
+            {
+                throw new ArgumentNullException(nameof(vsSolution));
+            }
+            if (logger == null)
+            {
+                throw new ArgumentNullException(nameof(logger));
+            }
+
+            this.activeSolutionBoundTracker = activeSolutionBoundTracker;
+            this.sonarQubeService = sonarQubeService;
+            this.workspace = workspace;
+            this.qualityProfileProvider = qualityProfileProvider;
+            this.vsSolution = vsSolution;
+            this.logger = logger;
+
+            this.activeSolutionBoundTracker = activeSolutionBoundTracker;
+            activeSolutionBoundTracker.SolutionBindingChanged += OnSolutionBindingChanged;
+
+            RefreshWorkflow(this.activeSolutionBoundTracker.CurrentConfiguration);
+        }
+
+        private void OnSolutionBindingChanged(object sender, ActiveSolutionBindingEventArgs e)
+        {
+            RefreshWorkflow(e.Configuration);
+        }
+
+        private void RefreshWorkflow(BindingConfiguration configuration)
+        {
+            this.sonarQubeIssueProvider?.Dispose();
+            this.currentWorklow?.Dispose();
+
+            switch (configuration.Mode)
+            {
+                case SonarLintMode.Standalone:
+                    this.currentWorklow = new SonarAnalyzerStandaloneWorkflow(this.workspace);
+                    break;
+
+                case SonarLintMode.LegacyConnected:
+                case SonarLintMode.Connected:
+                    this.sonarQubeIssueProvider = new SonarQubeIssuesProvider(sonarQubeService, configuration.Project.ProjectKey,
+                        new TimerFactory());
+                    var liveIssueFactory = new LiveIssueFactory(workspace, vsSolution);
+                    var suppressionHandler = new SuppressionHandler(liveIssueFactory, sonarQubeIssueProvider);
+
+                    this.currentWorklow = configuration.Mode == SonarLintMode.Connected
+                        ? (SonarAnalyzerWorkflowBase)new SonarAnalyzerConnectedWorkflow(this.workspace,
+                            this.qualityProfileProvider, configuration.Project, suppressionHandler)
+                        : new SonarAnalyzerLegacyConnectedWorkflow(this.workspace, suppressionHandler, this.logger);
+                    break;
+
+                default:
+                    this.currentWorklow = null;
+                    break;
+            }
+        }
+
+        #region IDisposable Support
+        private bool disposedValue = false;
+        public void Dispose()
+        {
+            if (disposedValue)
+            {
+                return;
+            }
+
+            this.activeSolutionBoundTracker.SolutionBindingChanged -= OnSolutionBindingChanged;
```

Comment:
Should perhaps also dispose the current workflow.

Reply:
You are right!